disperser/batcher/grpc/dispatcher.go

```diff
@@ -153,28 +156,35 @@ func getBlobMessage(blob *core.BlobMessage) (*node.Blob, error) {
 		lengthProofData.YA1 = blob.BlobHeader.LengthProof.Y.A1.Marshal()
 	}
 
-	quorumHeaders := make([]*node.BlobQuorumInfo, len(blob.BlobHeader.QuorumInfos))
+	quorumHeaders := make([]*node.BlobQuorumInfo, 0)
 
-	for i, header := range blob.BlobHeader.QuorumInfos {
-		quorumHeaders[i] = &node.BlobQuorumInfo{
+	for _, header := range blob.BlobHeader.QuorumInfos {
+		quorumHeaders = append(quorumHeaders, &node.BlobQuorumInfo{
 			QuorumId:           uint32(header.QuorumID),
 			AdversaryThreshold: uint32(header.AdversaryThreshold),
 			ChunkLength:        uint32(header.ChunkLength),
 			QuorumThreshold:    uint32(header.QuorumThreshold),
 			Ratelimit:          header.QuorumRate,
-		}
+		})
 	}
 
 	data, err := blob.Bundles.Serialize()
 	if err != nil {
 		return nil, err
 	}
-	bundles := make([]*node.Bundle, len(blob.Bundles))
+	bundles := make([]*node.Bundle, len(quorumHeaders))
 	// the ordering of quorums in bundles must be same as in quorumHeaders
 	for i, quorumHeader := range quorumHeaders {
 		quorum := quorumHeader.QuorumId
-		bundles[i] = &node.Bundle{
-			Chunks: data[quorum],
+		if _, ok := blob.Bundles[uint8(quorum)]; ok {
+			bundles[i] = &node.Bundle{
+				Chunks: data[quorum],
+			}
+		} else {
```

Comment:
Makes sure `len(quorumHeaders) == len(bundles)`
Bundles will be empty for quorums the operator is not part of